src/main/java/uk/gov/justice/hmpps/prison/service/ReferenceDomainService.java

```diff
@@ -132,11 +132,13 @@ public Optional<ReferenceCode> getReferenceCodeByDomainAndCode(final String doma
     }
 
     private void verifyReferenceDomain(final String domain) {
+        if (StringUtils.isBlank(domain)) throw new EntityNotFoundException("Reference domain not specified");
```

Comment:
decided it was safer to throw the same exception as thrown later in code rather than a different one